src/cli.ts

```diff
@@ -0,0 +1,72 @@
+import type * as http from 'node:http';
+import * as https from 'node:https';
+
+import { CHAINS } from '@api3/chains';
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+
+import { computeDappSpecificApi3ReaderProxyV1Address } from './proxy';
+
+const COMMON_COMMAND_ARGUMENTS = {
+  dappAlias: {
+    type: 'string',
+    demandOption: true,
+    describe: 'dApp alias as specified on https://docs.api3.org/dapps/oev-rewards/dapp-alias.html',
+  },
+  chainId: {
+    type: 'string',
+    demandOption: true,
+    describe: 'Chain ID',
+  },
+  dapiName: {
+    type: 'string',
+    demandOption: true,
+    describe: 'dAPI (data feed) name as it appears on https://market.api3.org/',
+  },
+} as const;
+
+const { dappAlias, chainId, dapiName } = COMMON_COMMAND_ARGUMENTS;
+
+async function requestToUrlReturns404(urlString: string): Promise<boolean> {
+  return new Promise((resolve, reject) => {
+    const request = https.get(urlString, (response: http.IncomingMessage) => {
+      response.resume();
+      resolve(response.statusCode === 404);
+    });
+    request.on('error', (error: Error) => {
+      reject(error);
+    });
+  });
+}
+
+// From https://github.com/api3dao/data-feeds/blob/main/packages/api3-market/src/utils/format.ts
+const slugify = (text: string) => text.toLowerCase().replaceAll(/[^\da-z-]+/g, '-');
+
+yargs(hideBin(process.argv))
+  .command(
+    'print-api3readerproxyv1-address',
+    'Prints the dApp-specific Api3ReaderProxyV1 address',
+    {
+      'dapp-alias': dappAlias,
+      'chain-id': chainId,
```

Comment:
This could also have been the chain alias. What I don't like about that is the user knows the chain ID but not necessarily the chain alias (and there is not a nice way of checking that), so this feels more user-friendly.